sky/packages/sky/lib/widgets/transitions.dart

```diff
@@ -167,3 +169,42 @@ class ColorTransition extends TransitionBase {
     );
   }
 }
+
+class SquashTransition extends TransitionBase {
+  SquashTransition({
+    Key key,
+    this.width,
+    this.height,
+    Duration duration,
+    AnimationPerformance performance,
+    Direction direction,
+    Function onDismissed,
+    Function onCompleted,
+    Widget child
+  }) : super(key: key,
+             duration: duration,
+             performance: performance,
+             direction: direction,
+             onDismissed: onDismissed,
+             onCompleted: onCompleted,
+             child: child);
+
+  AnimatedValue<double> width;
+  AnimatedValue<double> height;
+
+  void syncFields(SquashTransition updated) {
+    width = updated.width;
+    height = updated.height;
+    super.syncFields(updated);
+  }
+
+  double _maybe(AnimatedValue<double> x) => x != null ? x.value : null;
```

Comment:
This feels out-of-place here...  maybe make this into a free function at the top of this file?